File: java-app/src/test/java/com/demo/UserOperationsTest.java
package com.demo;

import com.azure.cosmos.*;
import com.azure.cosmos.models.*;
import org.junit.jupiter.api.*;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.*;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class UserOperationsTest {

    private CosmosClient client;
    private CosmosDatabase database;
    private CosmosContainer container;

    @BeforeAll
    public void setup() {
        String connectionString = System.getenv("COSMOSDB_CONNECTION_STRING");
        String databaseName = System.getenv("COSMOSDB_DATABASE_NAME");
        String containerName = System.getenv("COSMOSDB_CONTAINER_NAME");

        String endpoint = connectionString.split(";")[0].split("=")[1];
        System.out.println("endpoint 1 === " + endpoint);

        endpoint = "https://" + endpoint.split("://")[1];
        System.out.println("endpoint 2 === " + endpoint);

        String key = connectionString.split(";")[1].split("=")[1];
        System.out.println("key === " + key);

        try {
            this.client = new CosmosClientBuilder()
                    .endpoint(endpoint)
                    .key(key)
                    .gatewayMode()
                    .buildClient();

        } catch (Exception e) {
            System.out.println("create client object failed: " + e);
            throw e;
        }

        CosmosDatabaseResponse createDBResponse = null;
        try {
            createDBResponse = client.createDatabaseIfNotExists(databaseName);
        } catch (Exception e) {
            System.out.println("create database failed: " + e.getMessage());
            throw e;
        }

        this.database = client.getDatabase(createDBResponse.getProperties().getId());
        System.out.println("created database: " + database.getId());

        CosmosContainerResponse createContainerResponse = null;
        try {
            createContainerResponse = database.createContainerIfNotExists(containerName, "/id");
        } catch (Exception e) {
            System.out.println("create container failed: " + e.getMessage());
            throw e;
        }

        this.container = this.database.getContainer(createContainerResponse.getProperties().getId());
        System.out.println("created container: " + container.getId());
    }

    @AfterAll
    public void cleanup() {
        this.database.delete();
        client.close();
        System.out.println("deleted database: " + this.database.getId());

    }

    @Test
    public void testCreateUser() {
        UserOperations.User user = new UserOperations.User(UUID.randomUUID().toString(), "user42@example.com", true);
        assertDoesNotThrow(() -> UserOperations.createUser(this.container, user));
    }

    @Test
    public void testReadUser() {
        String userId = UUID.randomUUID().toString();
        UserOperations.User user = new UserOperations.User(userId, "user43@example.com", true);
        assertDoesNotThrow(() -> UserOperations.createUser(this.container, user));

        UserOperations.User readUser = assertDoesNotThrow(() -> UserOperations.readUser(container, userId));
        assertEquals(userId, readUser.getId());
        assertEquals("user43@example.com", readUser.getEmail());
        assertTrue(readUser.isActive());
    }
}

File: dotnet-app/UserOperationsTest.cs
using Microsoft.Azure.Cosmos;
using NUnit.Framework;

namespace dotnet_app.Tests
{
    [TestFixture]
    public class UserOperationsTest
    {
        private CosmosClient? client;
        private Database? database;
        private Container? container;

        [OneTimeSetUp]
        public async Task Setup()
        {
            string connectionString = Environment.GetEnvironmentVariable("COSMOSDB_CONNECTION_STRING") ?? throw new InvalidOperationException("COSMOSDB_CONNECTION_STRING environment variable is not set.");

            string databaseName = Environment.GetEnvironmentVariable("COSMOSDB_DATABASE_NAME") ?? throw new InvalidOperationException("COSMOSDB_DATABASE_NAME environment variable is not set.");

            string containerName = Environment.GetEnvironmentVariable("COSMOSDB_CONTAINER_NAME") ?? throw new InvalidOperationException("COSMOSDB_CONTAINER_NAME environment variable is not set.");

            connectionString = connectionString.Replace("http://", "https://");

            CosmosClientOptions options = new()
            {
                HttpClientFactory = () => new HttpClient(new HttpClientHandler()
                {
                    ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
                }),
                ConnectionMode = ConnectionMode.Gateway,
            };

            client = new CosmosClient(connectionString, options);
            database = await client.CreateDatabaseIfNotExistsAsync(databaseName);
            container = await database.CreateContainerIfNotExistsAsync(containerName, "/id");

            Console.WriteLine("Created database: " + database.Id);
            Console.WriteLine("Created container: " + container.Id);
        }

        [OneTimeTearDown]
        public async Task Cleanup()
        {
            if (database != null)
            {
                await database.DeleteAsync();
                Console.WriteLine("Deleted database: " + database.Id);
            }
            client?.Dispose();
        }

        [Test]
        public async Task TestCreateUser()
        {
            User user = new()
            {
                id = Guid.NewGuid().ToString(),
                email = "user42@example.com",
                active = true
            };

            if (container == null)
            {
                throw new InvalidOperationException("Container is not initialized.");
            }
            Assert.DoesNotThrowAsync(async () => await UserOperations.CreateUserAsync(container, user));
        }

        [Test]
        public async Task TestReadUser()
        {
            string userId = Guid.NewGuid().ToString();

            User user = new()
            {
                id = userId,
                email = "user43@example.com",
                active = true
            };

            if (container == null)
            {
                throw new InvalidOperationException("Container is not initialized.");
            }

            Assert.DoesNotThrowAsync(async () => await UserOperations.CreateUserAsync(container, user));

            User readUser = await UserOperations.ReadUserAsync(container, userId);
            Assert.AreEqual(userId, readUser.id);
            Assert.AreEqual("user43@example.com", readUser.email);
            Assert.IsTrue(readUser.active);
        }
    }
}